### PR TITLE
Start replacing YUI events

### PR DESCRIPTION
**Pull request by huwshimi**
Replace YUI events for:
- Y.on
- Y.once
- Y.fire
- db.fire('update')

### CODE REVIEW

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -193,13 +193,6 @@ YUI.add('juju-gui', function(Y) {
         help: 'Display this help',
         label: 'Shift + ?'
       },
-      'A-e': {
-        callback: function(evt) {
-          this.fire('navigateTo', { url: '/:gui:/' });
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/app.js]: This is not used anymore.

File: jujugui/static/gui/src/app/app.js [hunk 2]
```diff
@@ -233,14 +226,6 @@ YUI.add('juju-gui', function(Y) {
         },
         help: 'Export the model',
         label: 'Shift + d'
-      },
-
-      'C-S-d': {
-        callback: function(evt) {
-          Y.fire('saveWebsocketLog');
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/app.js]: We don't use this event anymore.

File: jujugui/static/gui/src/app/app.js [hunk 3]
```diff
@@ -561,10 +546,6 @@ YUI.add('juju-gui', function(Y) {
       // Listen for window unloads and trigger the unloadWindow function.
       window.onbeforeunload = views.utils.unloadWindow.bind(this);
 
-      this.on('*:navigateTo', function(e) {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/app.js]: Unused

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7175/

**Comment by huwshimi** [comment]
QA:
- add a couple of apps and relate them
- deploy
- switch to another model
- these actions should all work if the db updates are fired.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7176/

File: jujugui/static/gui/src/app/app.js [hunk 4]
```diff
@@ -646,9 +628,10 @@ YUI.add('juju-gui', function(Y) {
 
       // When someone wants a charm to be deployed they fire an event and we
       // show the charm panel to configure/deploy the service.
-      Y.on('initiateDeploy', function(charm, ghostAttributes) {
-        this.deployService(charm, ghostAttributes);
-      }, this);
+      this._onInitiateDeploy = e => {
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/app.js]: `e` -> `evt`

File: jujugui/static/gui/src/app/models/models.js [hunk 5]
```diff
@@ -2178,6 +2178,15 @@ YUI.add('juju-models', function(Y) {
     },
 
     /**
+      Dispatch an event.
+
+     @method fireEvent
+    */
```
Comment (Lukewh) [inline jujugui/static/gui/src/app/models/models.js]: add docstring for `event`?

**Comment by Lukewh** [comment]
## QA OK

Switching between models and using the branch to do some deploys seems to be good! BYe-UI, amiright?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7196/

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge/2287

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**